### PR TITLE
[bzip2] fixes for building release-only

### LINKED ISSUE #12799: [bzip2] release-only build failure on Windows
**Host Environment**

- OS: Windows 10 2019 from GitHub Actions
- Compiler: Latest MSVC 14.26.xxxxx

**To Reproduce**

`./vcpkg install bzip2:x64-windows-static-release --clean-after-build --overlay-triplets=triplets_overlay`

(or any package that depends on `bzip2`, like `qt5-base`)

- Contents of `triplets_overlay/x64-windows-static-release.cmake`:

    ```
    set(VCPKG_TARGET_ARCHITECTURE x64)
    set(VCPKG_CRT_LINKAGE static)
    set(VCPKG_LIBRARY_LINKAGE static)
    
    set(VCPKG_BUILD_TYPE release)
    ```

**Failure logs** 

```
Starting package 2/16: bzip2:x64-windows-static-release
Building package bzip2[core]:x64-windows-static-release...
Could not locate cached archive: C:\Users\runneradmin\AppData\Local\vcpkg\archives\2c\2c926ea673e5e047cbce2782a0a8d80dcb8aaa26.zip
-- [OVERLAY] Loading triplet configuration from: D:\a\qBittorrent\qBittorrent\triplets_overlay\x64-windows-static-release.cmake
-- Downloading https://github.com/past-due/bzip2-mirror/releases/download/v1.0.6/bzip2-1.0.6.tar.gz...
-- Extracting source C:/qbt_tools/vcpkg/downloads/bzip2-1.0.6.tar.gz
-- Applying patch fix-import-export-macros.patch
-- Applying patch fix-windows-include.patch
-- Using source at C:/qbt_tools/vcpkg/buildtrees/bzip2/src/1.0.6-8838f21f8e.clean
-- Configuring x64-windows-static-release
-- Building x64-windows-static-release-rel
-- Fixing pkgconfig
-- Acquiring MSYS Packages...
-- Acquiring MSYS Packages... OK
-- Fixing pkgconfig - release
-- Checking file: C:/qbt_tools/vcpkg/packages/bzip2_x64-windows-static-release/lib/pkgconfig/bzip2.pc
-- Fixing pkgconfig - debug
-- Checking file: C:/qbt_tools/vcpkg/packages/bzip2_x64-windows-static-release/debug/lib/pkgconfig/bzip2.pc
-- CHECK_LIB_bz2d_DEBUG:CHECK_LIB_bz2d_DEBUG-NOTFOUND
CMake Error at scripts/cmake/vcpkg_fixup_pkgconfig.cmake:239 (message):
  Library "bz2d" was not found! If it is a system library use the
  SYSTEM_LIBRARIES parameter for the vcpkg_fixup_pkgconfig call! Otherwise,
  correct the *.pc file
Call Stack (most recent call first):
  scripts/cmake/vcpkg_fixup_pkgconfig.cmake:348 (vcpkg_fixup_pkgconfig_check_files)
  ports/bzip2/portfile.cmake:51 (vcpkg_fixup_pkgconfig)
  scripts/ports.cmake:79 (include)


Error: Building package bzip2:x64-windows-static-release failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: bzip2:x64-windows-static-release
  Vcpkg version: 2020.06.15-nohash

Additionally, attach any relevant sections from the log files above.
```

**Additional context**

This is different from https://github.com/microsoft/vcpkg/issues/12689.

Looks like the build process expects to always find stuff related to the debug build, even if just building release (and probably vice-versa).

### PR DESCRIPTION
**Pull request by cenit**
Restore the possibility to build bzip2 with a release-only triplet.
I'd also suggest to move one of the CI pipelines to test ports for this configuration, which is very common in production.
Closes #12799

### CODE REVIEW

**Comment by cenit** [comment]
EOL is now LF only, sorry for the very noisy diff. Luckily GitHub has an option in the diff view to hide whitespace-only changes

File: ports/bzip2/portfile.cmake [hunk 1]
```diff
@@ -1,51 +1,53 @@
...
+
+file(READ "${CURRENT_PACKAGES_DIR}/include/bzlib.h" BZLIB_H)
+if(VCPKG_LIBRARY_LINKAGE STREQUAL "static")
+    string(REPLACE "defined(BZ_IMPORT)" "0" BZLIB_H "${BZLIB_H}")
+else()
+    string(REPLACE "defined(BZ_IMPORT)" "1" BZLIB_H "${BZLIB_H}")
+endif()
+file(WRITE "${CURRENT_PACKAGES_DIR}/include/bzlib.h" "${BZLIB_H}")
+
+if (NOT VCPKG_BUILD_TYPE OR VCPKG_BUILD_TYPE STREQUAL "debug")
```
Comment (ras0219) [inline ports/bzip2/portfile.cmake:38]:
```suggestion
if (NOT VCPKG_BUILD_TYPE OR VCPKG_BUILD_TYPE STREQUAL "release")
```

Comment (cenit) [reply ports/bzip2/portfile.cmake]: Thank you!!!! I did a big mistake while preparing the PR!

**Comment by LilyWangL** [comment]
The error on baseline will be fixed in PR #12766.

**Comment by Neumann-A** [comment]
@cenit: please stop mixing port fixes with smaller script fixes. Your are invalidating all the cached archives which use `vcpkg_fixup_pkgconfig()` and don't depend on `bzip2`. Although you might not see the benefit of `message(STATUS "Checking file: ${_file}")` since you probably only dealt with ports installing only a few `*.pc` files it is quite important to know where the error occurred if you have to deal with a port installing 20+ `*.pc` files

**Comment by cenit** [comment]
That's a problem for port creators, not port users. For the same reasons there is no output for vcpkg_fixup_cmake_targets or many others.
We should assume that when you install the port, it should just work, and output should be minimal as it always has been

**Comment by c72578** [comment]
> 
> We should assume that when you install the port, it should just work

This PR here already shows the opposite of such an assumption

**Comment by cenit** [comment]
> This PR here already shows the opposite of such an assumption

again, that's not a reason to fill up user screen with unnecessary debug information.
Such problems should be detected during review and from CI

**Comment by Neumann-A** [comment]
@cenit with that logic vcpkg should not output anything. The messages are meant as progress messages not debug info. If you would rather like to see 'fixing pkgconfig x of y' I can make that change. I would also like to see some progress messages for fixup_cmake because ports like qt and vtk take a long time to fixup due to the amount of small file io.

**Comment by cenit** [comment]
> with that logic vcpkg should not output anything

tbh I'd love it :)
Output only in case of errors, otherwise just a sequence of 
"Starting package 1/97"
"Finishing package 1/97"
....
"Starting package 97/97"
"Finishing package 97/97"
such a lovely output 😄

**Comment by AenBleidd** [comment]
Maybe just a global verbosity flag should be added for vcpkg? Then every port user can control amount of information they'd wish to see...

**Comment by cenit** [comment]
of course, on files (with a fixed standardized syntax) of course you can write as much as you want!

**Comment by AenBleidd** [comment]
@LilyWangL, any news here? All CI fails looks like not really related to this particular PR...

**Comment by past-due** [comment]
I've also hit this issue, and am eager to see this fix merged.

**Comment by LilyWangL** [comment]
The CI fails is not related with this PR, it will be fixed by PR #12856.

**Comment by gitcop-devel** [comment]
@cenit, please follow common git commit practices:
> Make only one change per commit and avoid including unrelated changes

The changes to the file scripts/cmake/vcpkg_fixup_pkgconfig.cmake are unrelated to "[bzip2] fixes for building release-only" (commit a3e2935e71ba2322bc7e715604eff41a70be450b)

**Comment by AenBleidd** [comment]
@gitcop-devel, this repo contains examples that contradict this rule ;)

**Pull request closed without merging**